nomad/state/events_state_store.go

```diff
@@ -0,0 +1,47 @@
+package state
+
+import (
+	"fmt"
+
+	memdb "github.com/hashicorp/go-memdb"
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+// addNodeEvent is a function which wraps upsertNodeEvent
```

Comment:
Comment doesn't match function name. Comments should describe the behavior not the implementation detail ( what happens if upsertNodeEvent gets renamed, this comment is no longer valid). Instead:

` AddNodeEvent adds the given even to the specified node. The Nodes event list will be truncated if necessary to avoid an unbounded number of events.`